data/maps/SP_Scenarios/Along_The_River.wmf/scripting/init.lua

```diff
@@ -0,0 +1,404 @@
+      sleep(30000)
+      map:get_field(31,1).terr = "summer_mountain2"
+      map:get_field(30,142).terr = "summer_mountain4"
+      map:get_field(30,142).terd = "summer_mountain1"
+      p1:hide_fields(port:region(6))
+      local pirate_fortress = map:get_field(55, 60)
+      if pirate_fortress.immovable and pirate_fortress.owner == p5 then
+         -- Soldier level must match with level in initial_conditions.
+         pirate_fortress.immovable:set_soldiers({0,2,0,2}, 8)
+      end
+      while not immovable_has_name(port.immovable, "barbarians_port") do sleep(2039) end
+      o_build_port.done = true
+      map:get_field(30,143).terd = "summer_mountain2"
+      map:get_field(29,143).terr = "summer_mountain3"
+   end)
+
+   -- Conquer pirate barbarians_sentry and build warehouse.
+   run(function()
+      local pirate_sentry = map:get_field(59, 89)
+      while not p1:seen_field(pirate_sentry) do
+         sleep(2011)
+         -- Ai will dismantle sentry cause it can’t be connected to a WH so check and rebuild
+         local sentry_immovable = pirate_sentry.immovable
+         if not sentry_immovable or sentry_immovable.descr.name ~= "barbarians_sentry" then
+            prefilled_buildings(p5, {"barbarians_sentry", 59, 89, soldiers = {[{0,2,0,2}] = 2 }})
+         end
+      end
+      scroll_to_field(pirate_sentry)
+      campaign_message_box(briefing_conquer_pirate_sentry)
+      local o_conquer_pirate_sentry = add_campaign_objective(obj_conquer_pirate_sentry)
+
+      while pirate_sentry.owner == p5 do sleep(997) end
+      p1:place_flag(map:get_field(53, 87), true)
+      map:get_field(53, 87).immovable:remove()
+      p1:reveal_fields(warehouse:region(6))
+      scroll_to_field(warehouse)
+      campaign_message_box(briefing_found_pirate_treasure)
+      o_build_warehouse = add_campaign_objective(obj_build_warehouse)
+      o_conquer_pirate_sentry.done = true
+      run(function()
+         sleep(30000)
+         p1:hide_fields(warehouse:region(6))
+      end)
+      transport_wares(p1, hq, "gold_ore", 25, true, "tribes/buildings/warehouses/barbarians/headquarters_interim/idle_00.png")
+      while not o_build_warehouse.done do
+         sleep(2053)
+         for i, field in pairs(map:get_field(53,92):region(8)) do
+            if immovable_has_name(field.immovable, "barbarians_warehouse") then
+               warehouse = field
+               o_build_warehouse.done = true
+               break
+            end
+         end
+      end
+   end)
+
+   while not (obj_done(o_build_warehouse) and obj_done(o_build_port)) do sleep(2063) end
+
+   -- =======================================================================
+   -- Bring shovels and remove the swamps.
+   -- =======================================================================
+
+   -- Wait for shovels in warehouse and hand them (done in "wares_collected_at_field" function)out.
+   run(function()
+      push_textdomain("map_along_the_river.wmf")
```

Comment:
same here, the `while not` loops extend the textdomain scope too much. Instead, each of the two translated lines in this function should have a `push/pop` call directly above and below, as in
```lua
push_textdomain("map_along_the_river.wmf")
send_to_inbox(p1, _"Shovels, ...)
pop_textdomain()
```